### PR TITLE
Validated geoms using osgeo.ogr and shapely

### LINKED ISSUE #1071: Fix records (78 of them) with self-intersecting geometries
The following returns 78 records that yield `Self-intersection at or near point...` or `Ring Self-intersection at or near point...` according to GDAL:

```
import mapzen.whosonfirst.utils
import osgeo.ogr
import geojson
import sys

if __name__ == "__main__":

    data = sys.argv[1]
    crawl = mapzen.whosonfirst.utils.crawl(data, inflate=True)

    for f in crawl:

	poly = osgeo.ogr.CreateGeometryFromJson(geojson.dumps(f['geometry']))
	valid = poly.IsValid()

	if not valid:
            print "%s %s" % (f['properties']['wof:id'], valid)
```

The complete list is attached but you can see an example here:

https://whosonfirst.mapzen.com/spelunker/id/85766257/

cc @simonw

[report.txt](https://github.com/whosonfirst-data/whosonfirst-data/files/1624705/report.txt)

### PR DESCRIPTION
**Pull request by stepps00**
WIll fix #1071.

This PR corrects any "invalid" geometries (self-intersections, intersecting rings, etc) for any record in the whosonfirst-data repo. Commits include:

- [#1](https://github.com/whosonfirst-data/whosonfirst-data/pull/1293/commits/cbd2accedee702f37bb49eb8af8ab5d82de0de23): Iterates over the list of invalid ids, updating geometries using the `buffer(0.0)` trick. Script [here](https://github.com/whosonfirst/whosonfirst-cookbook/blob/master/scripts/1293_geom_validator.py).
- [#2](https://github.com/whosonfirst-data/whosonfirst-data/pull/1293/commits/c98d0d850f87afb1c0739df599604d60447720e1): Updates the Pongola records, superseding the three existing records into a single record and updating the [Pongola record](https://spelunker.whosonfirst.org/id/101927443/) that has an invalid geometry.
- [#3](https://github.com/whosonfirst-data/whosonfirst-data/pull/1293/commits/2288c9aade29e05b8f02d7a27027582e1a7dd1f6): Occassionally the `buffer(0.0)` trick results in an empty polygon. This commit updates the records that were not updated in the first commit, hand-crafting new geometries for "leftover" records that have invalid geometries.

Full list of ids tagged with "invalid" geometries*: [list](https://github.com/whosonfirst-data/whosonfirst-data/files/2325566/all_invalid_ids.txt)

*Note that some of the ids in this list were recently updated in recent PRs and no longer contain invalid polygon geometries.

### CODE REVIEW

**Comment by nvkelso** [comment]
@stepps00 in https://github.com/whosonfirst-data/whosonfirst-data/issues/1071#issuecomment-415904700 you said 1455 places were affected but the list in this PR description has 2140. Did the methodology change and how?

**Comment by stepps00** [comment]
@nvkelso - there are 1455 records with invalid geometries, plus three records updated from supersedes work in https://github.com/whosonfirst-data/whosonfirst-data/pull/1293/commits/c98d0d850f87afb1c0739df599604d60447720e1 (= total count of 1458 in this PR).

The list in the comment above includes all ids that were found to have invalid geometries at some point, but recent merge work/duplicate work/work in disputed areas updated many of them.

In short, that count of 1455 is correct.

**Comment by stepps00** [comment]
Noting that occasionally `campus` geometries from `woedb` with self-intersections are getting smaller, with outer rings being removed.

Example:

https://github.com/whosonfirst-data/whosonfirst-data/blob/2288c9aade29e05b8f02d7a27027582e1a7dd1f6/data/102/530/841/102530841.geojson

vs

https://spelunker.whosonfirst.org/id/102530841/

Going forward, this could be fixed by converting the geometry to a wkt, clipping each geometry at the intersection point, calculating outer rings, merging, then dissolving (perhaps there is a simpler way).
